internal/connector/connector.go

```diff
@@ -77,13 +80,36 @@ type KubernetesOptions struct {
 
 // connector stores all the dependencies for the connector operations.
 type connector struct {
-	k8s         *kubernetes.Kubernetes
-	client      *api.Client
+	k8s         kubeClient
+	client      apiClient
 	destination *api.Destination
 	certCache   *CertCache
 	options     Options
 }
 
+type apiClient interface {
```

Comment:
What advantage does creating a new interface for this give us over using the api.Client?

Reply:
Good question! I made these interface so that we can use a fake in tests. That way I can easily write a test to show that this loop behave correctly when there are errors from the API. 

We could have used a full `httptest.Server` to return error messages, but I think that ends up being a bit more work. A fakeClient was a bit easier.

A secondary benefit is that we can easily see which API calls the connector is using from this interface, without having to trace all of the code.